syft/pkg/cataloger/cpe.go

```diff
@@ -80,6 +80,11 @@ func candidateVendors(p pkg.Package) []string {
 	switch p.Language {
 	case pkg.Python:
 		vendors = append(vendors, fmt.Sprintf("python-%s", p.Name))
+		if strings.Contains(p.Name, "-") {
```

Comment:
Why not just always do the replacement? Seems like negligible performance cost, but maybe something I'm missing too...my thinking is it'd just simplify this block
```
vendors = append(vendors.fmt.Sprintf("python_%s", strings.ReplaceAll(p.Name, "-", "_")))
```

Reply:
Also are there some language distributions that need a hyphen? Or is it inconsistent across languages in NVD?

Reply:
actually, looks like adjusting for [this comment](https://github.com/anchore/syft/pull/397#discussion_r616235716) will handle this case since vendors are duplicate of the product candidates.